src/globus_cli/termio/terminal_info.py

```diff
@@ -0,0 +1,40 @@
+"""
+Global interface for managing accessing and modifying terminal info.
+
+Some printers and formatters within this termio module are concerned with text wrapping.
+To do this effectively, they need to know where exactly to start wrapping text.
+"""
+
+import contextlib
+import shutil
+import typing as t
+
+__all__ = ("TERM_INFO",)
+
+
+class VirtualTerminalInfo:
+    MIN_COLUMNS = 6
+
+    def __init__(self) -> None:
+        self._column_delta = 0
+        cols, rows = shutil.get_terminal_size(fallback=(80, 20))
+        self._base_columns = cols if cols < 100 else int(0.8 * cols)
+        self._base_rows = rows
+
+    @contextlib.contextmanager
+    def indented(self, size: int) -> t.Iterator[None]:
+        """
+        Context manager to temporarily decrease the available width for text wrapping.
+        """
+
+        self._column_delta -= size
+        yield
+        self._column_delta += size
+
+    @property
+    def columns(self) -> int:
+        computed_columns = self._base_columns + self._column_delta
+        return max(self.MIN_COLUMNS, computed_columns)
+
+
+TERM_INFO = VirtualTerminalInfo()
```

Comment:
Can these refinements, to use `VirtualTerminalInfo` to capture this functionality and use it in `_item_wrapper` for fields, be separated into a distinct PR?

I think these are good and separable improvements. Having them isolated will make review much easier.

Reply:
Help me understand the scope you're envisioning?

I bundled them all toegether because:
1. The `VirtualTerminalInfo` refactor is justified by a need to be able to indent your terminal.
1. The ability to indent makes sense when you need nested multiline string formatters
1. Multiline string formatters make sense when you add a command that needs nested multiline string formatting
1. That command is `globus auth scope show`.

I don't see anything besides item (4) that justifies itself or establishes why we would shape a thing a certain way. But maybe I'm missing it?

Reply:
I'm looking at this from the perspective of the PR reviewer, and finding that the semantic change -- addition of a new command -- is harder to review because the PR itself includes a body of refactoring work.

If I can focus on reviewing the refactor itself, I can provide better feedback.
It may be driven by this feature, but my opinion is that the work can be trivially split off and that it therefore should be.

Reply:
We discussed and agreed:
1. This really isn't important.
2. A rebase of this feature PR is enough work to make us decide against it.

Agreement reached! :tada: